src/initialization/InitDistribution.cpp

```diff
@@ -171,163 +260,144 @@ namespace impactx
-          pp_dist.get("sigmaPx", sigpx);
-          pp_dist.get("sigmaPy", sigpy);
-          pp_dist.get("sigmaPt", sigpt);
-          pp_dist.query("muxpx", muxpx);
-          pp_dist.query("muypy", muypy);
-          pp_dist.query("mutpt", mutpt);
-
-          distribution::KnownDistributions const kvDist(distribution::KVdist(
-            sigx, sigy, sigt,
-            sigpx, sigpy, sigpt,
-            muxpx, muypy, mutpt));
-
-          add_particles(bunch_charge, kvDist, npart);
-
-        } else if (distribution_type == "kurth4d") {
-          amrex::ParticleReal sigx,sigy,sigt,sigpx,sigpy,sigpt;
-          amrex::ParticleReal muxpx = 0.0, muypy = 0.0, mutpt = 0.0;
-          pp_dist.get("sigmaX", sigx);
-          pp_dist.get("sigmaY", sigy);
-          pp_dist.get("sigmaT", sigt);
-          pp_dist.get("sigmaPx", sigpx);
-          pp_dist.get("sigmaPy", sigpy);
-          pp_dist.get("sigmaPt", sigpt);
-          pp_dist.query("muxpx", muxpx);
-          pp_dist.query("muypy", muypy);
-          pp_dist.query("mutpt", mutpt);
-
-          distribution::KnownDistributions const kurth4D(distribution::Kurth4D(
-            sigx, sigy, sigt,
-            sigpx, sigpy, sigpt,
-            muxpx, muypy, mutpt));
-
-          add_particles(bunch_charge, kurth4D, npart);
-        } else if (distribution_type == "semigaussian") {
-          amrex::ParticleReal sigx,sigy,sigt,sigpx,sigpy,sigpt;
-          amrex::ParticleReal muxpx = 0.0, muypy = 0.0, mutpt = 0.0;
-          pp_dist.get("sigmaX", sigx);
-          pp_dist.get("sigmaY", sigy);
-          pp_dist.get("sigmaT", sigt);
-          pp_dist.get("sigmaPx", sigpx);
-          pp_dist.get("sigmaPy", sigpy);
-          pp_dist.get("sigmaPt", sigpt);
-          pp_dist.query("muxpx", muxpx);
-          pp_dist.query("muypy", muypy);
-          pp_dist.query("mutpt", mutpt);
-
-          distribution::KnownDistributions const semigaussian(distribution::Semigaussian(
-            sigx, sigy, sigt,
-            sigpx, sigpy, sigpt,
-            muxpx, muypy, mutpt));
-
-          add_particles(bunch_charge, semigaussian, npart);
-
-        } else if (distribution_type == "triangle") {
-          amrex::ParticleReal sigx, sigy, sigt, sigpx, sigpy, sigpt;
-          amrex::ParticleReal muxpx = 0.0, muypy = 0.0, mutpt = 0.0;
-          pp_dist.get("sigmaX", sigx);
-          pp_dist.get("sigmaY", sigy);
-          pp_dist.get("sigmaT", sigt);
-          pp_dist.get("sigmaPx", sigpx);
-          pp_dist.get("sigmaPy", sigpy);
-          pp_dist.get("sigmaPt", sigpt);
-          pp_dist.query("muxpx", muxpx);
-          pp_dist.query("muypy", muypy);
-          pp_dist.query("mutpt", mutpt);
-
-          distribution::KnownDistributions const triangle(distribution::Triangle(
-            sigx, sigy, sigt,
-            sigpx, sigpy, sigpt,
-            muxpx, muypy, mutpt));
-
-          add_particles(bunch_charge, triangle, npart);
+        std::string base_dist_type = distribution_type;
```

Comment:
The "has `_`" is not as robust as I would prefer. Can we instead change the logic to "ends in `_from_cs` || `_from_twiss`"?